### PR TITLE
Update dev signup form

### PR DESCRIPTION
**Pull request by rowannekabalan**
## What does this change?

This updates the Developer Registration form in line with the changes proposed by licensing in this [https://docs.google.com/document/d/1w8Er8ZXG_aKe4rmpUHmYgDtGnQQsWx4Zr9ezKqMWRxM/edit?hl=en-GB&forcehl=1](document).

The objective of these changes is to capture more meaningful info on how developers intend to use our keys, recognise opportunities for commercialisation as well as recognise misuse of free developer keys.

We updated other forms in Bonobo with the same changes for the sake of consistency.

We also updated the Key and User models to include the new fields being captured and updated all related CRUD functions.

## How to test

This is deployed and tested on CODE. Can be tested locally as well by running Bonobo as per instructions in the README file.

## How can we measure success?

Licensing are happy!

## Have we considered potential risks?

This is a medium risk change as it involves model changes as well as UI changes.

## Images
Before:
Screenshot 2022-05-16 at 11 23 11

After:
Screenshot 2022-05-16 at 11 26 10

## Accessibility

- [ ] [Tested with screen reader](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#screen-readers)
- [ ] [Navigable with keyboard](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#keyboard-navigation)
- [ ] [Colour contrast passed](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#colour-contrast)
- [X] [The change doesn't use only colour to convey meaning](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#use-of-colour)

### CODE REVIEW

File: app/models/models.scala [hunk 1]
```diff
@@ -12,29 +12,30 @@ case class BonoboUser(
...
     val additionalInfo = AdditionalUserInfo(DateTime.now(), DeveloperRegistration)
-    new BonoboUser(id, formData.email, formData.name, formData.companyName, formData.companyUrl, additionalInfo, List.empty)
+    new BonoboUser(id, formData.email, formData.name, formData.companyName, formData.companyUrl, formData.companySector, additionalInfo, List.empty)
   }
   /* Method used when using the commercial form for creating a user */
   def apply(formData: CommercialRequestKeyFormData): BonoboUser = {
     val additionalInfo = AdditionalUserInfo(DateTime.now(), CommercialRegistration, Some(formData.businessArea), Some(formData.monthlyUsers.toString), Some(formData.commercialModel), Some(formData.content), Some(formData.contentFormat), Some(formData.articlesPerDay.toString))
-    new BonoboUser(java.util.UUID.randomUUID().toString, formData.email, formData.name, Some(formData.companyName), Some(formData.companyUrl), additionalInfo, List.empty)
+    new BonoboUser(java.util.UUID.randomUUID().toString, formData.email, formData.name, Some(formData.companyName), Some(formData.companyUrl), None, additionalInfo, List.empty)
```
Comment (JustinPinner) [inline app/models/models.scala:38]: Might be worth dropping a comment to explain why `companySector` is `None` for commercial form submissions.

**Comment by JustinPinner** [comment]
This is looking good. It's a shame the templating layer gives rise to so much duplication of effort. Never mind though, when this is ready for review we'll give it one last 👀 and award it a 👍

File: app/models/models.scala (hunk 1 above)
Comment (rowannekabalan) [reply app/models/models.scala:39]: Done 👍

**Review by JustinPinner — APPROVED** [review]
LGTM 👍 Let's deploy it in the A.M.?

**Pull request merged**